### PR TITLE
Mtc unit test

### PR DESCRIPTION
**Pull request by pnrobinson**
Cleans up the p value/multiple testing correction classes

### CODE REVIEW

**Comment by pnrobinson** [comment]
This commit is a refactoring and simplification of code for over-representation testing and multiple testing correction that originated with the Ontologizer. I have brought the code nearly up to date (i.e., Java 1.4/1.5 -> 1.8) and removed parts of if that were mainly designed to interoperate with the Ontologizer GUI etc. We can now use the code to start to implement more of the Ontologizer algorithms in phenol: (i) other enrichment procedures (ii) remaining MTCs (iii) MGSA and (iv) in the future some statistical analysis that is more HP/MP specific.
I will merge this PR into develop soon, since it does not have any intersections outside of the GO stuff, but it would be great if @julesjacobsen @cmungall or others wanted to have a look at this module at some point and criticize overall architecture etc.

File: phenol-analysis/src/main/java/org/monarchinitiative/phenol/analysis/StudySet.java [hunk 1]
```diff
@@ -63,55 +65,54 @@ public StudySet(Set<TermId> genes, String name, AssociationContainer association
...
+  public Set<TermId> getAnnotatingTermIds() {
     return this.annotationMap.keySet();
   }
 
   /**
    * return the genes annotating any gene in this study set.
    *
-   * @param goId a GeneOntology id
-   * @return TermAnnotations object with all of the genes that annotate to goId
+   * @param goId a GO or HP id
```
Comment (julesjacobsen) [inline phenol-analysis/src/main/java/org/monarchinitiative/phenol/analysis/StudySet.java]: Should the param not be ```termId```

File: phenol-analysis/src/main/java/org/monarchinitiative/phenol/stats/Item2PValue.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      this.p_raw =p;
+      this.p_adjusted=p;
+  }
+
+  public T getItem() {
+    return this.item;
+  }
+
+
+  public void setAdjustedPValue(Double padj) {
```
Comment (julesjacobsen) [inline phenol-analysis/src/main/java/org/monarchinitiative/phenol/stats/Item2PValue.java]: Why have two ```setAdjustedPValue()```? You're using a boxed and un-boxed. You should prefer the unboxed.

File: phenol-analysis/src/main/java/org/monarchinitiative/phenol/analysis/StudySet.java (hunk 1 above)
Comment (pnrobinson) [reply phenol-analysis/src/main/java/org/monarchinitiative/phenol/analysis/StudySet.java]: hmm, I think the Javadoc params are OK.

File: phenol-analysis/src/main/java/org/monarchinitiative/phenol/stats/Item2PValue.java (hunk 2 above)
Comment (pnrobinson) [reply phenol-analysis/src/main/java/org/monarchinitiative/phenol/stats/Item2PValue.java]: upps, thanks -- corrected

**Comment by pnrobinson** [comment]
Codacy says: "Use explicit scoping instead of the default package private level" however, this was a conscious choice rather than forgetting something (I tried to get Codacy to change this...to no effect). I think it is OK.

**Pull request merged**